lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1143,7 +1143,7 @@ public function walkJoin($join)
                 $dqlAlias   = $joinDeclaration->aliasIdentificationVariable;
                 $tableAlias = $this->getSQLTableAlias($class->table['name'], $dqlAlias);
                 $condition = '(' . $this->walkConditionalExpression($join->conditionalExpression) . ')';
-                $isUnconditionalJoin = empty($conditions);
+                $isUnconditionalJoin = empty($condition);
```

Comment:
@jvasseur yes, that's not an issue though, as the code was cleaned up on `master`